docs/modules/ROOT/pages/index.adoc

```diff
@@ -27,41 +27,101 @@ In your `pom.xml` file, add:
+
+In other cases, the extension uses structured logging, via JSON serialization.
+There are two supported structured formats:
+
+* The `nested` serialization is the default format of Splunk HEC Java client and defines the name of some pre-defined metadata. Combined with `quarkus.log.handler.splunk.format=%s%e` it also support log messages that are themselves JSON.
+* The `flat` serialization is a simpler and more format, also used by the https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#splunk-hec[OpenTelemetry Splunk HEC exporter].
+
+Some metadata can be indexed by Splunk, see link:++https://docs.splunk.com/Splexicon:Indexedfield++[indexed fields].
+The default `_json` source type indexes metadata passed in the `fields` object.
+
+The extension provides the support of the resolution of MDC scoped properties, as defined in https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html/configuration_guide/logging_with_jboss_eap#log_formatters[JBoss supported formatters].
+
+[%header,cols="h,a,a"]
+|===
+|Serialization format
+|`nested`
+|`flat`
+
+|HEC metadata
+2+|`time` and `host` are always sent.
+`source`, `sourcetype`, `index` are sent if not empty.
+
+|Pre-defined metadata
+|Only `event.severity` is sent by default.
+Other metadata can be added:
+
+* `event.thread` via `quarkus.log.handler.splunk.include-thread-name`
+* `event.exception` via `quarkus.log.handler.splunk.include-exception`
+* `event.logger` via `quarkus.log.handler.splunk.include-logger-name`
+|Only `fields.severity` is sent.
+The metadata name can be customized via `quarkus.log.handler.splunk.metadataSeverityFieldName`
+|MDC properties
+|Passed via `event.properties`
+|Passed via `fields`
 
-The extension also provides the support of the resolution of MDC scoped properties, as defined in https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html/configuration_guide/logging_with_jboss_eap#log_formatters[JBoss supported formatters].
-MDC key/values are sent as link:++https://docs.splunk.com/Splexicon:Indexedfield++[indexed fields] in the event.
+|Static metadata
+2+|Passed via `fields`
 
-A structured log event looks like:
+|===
 
+.Nested serialization example
 [source,json]
 ----
 {
-  "time": "timestamp",
+  "time": "1673001538.042",
   "host": "hostname",
   "source": "mysource",
-  "sourcetype": "mysourcetype",
+  "sourcetype": "_json",
   "index": "myindex",
   "event": {
-    "message": "2021-10-07 ERROR The log message",
+    "message": "2023-01-06 ERROR The log message",
     "logger": "com.acme.MyClass",
     "severity": "ERROR",
     "exception": "java.lang.NullPointerException",
     "properties": {
       "mdc-key": "mdc-value"
     }
+  },
+  "fields": {
+    "key": "static-value"
+  }
+}
+----
+
+.Flat serialization example
```

Comment:
I have the impression that the flat format include less "interesting metadata", (logger, exception ...) than the nested format. 
It means that everyone will have to provide its own metadata through the fields structure ?
Can't the extension provides it for the user by reusing the current pre-defined metadata?

Reply:
Yes we can actually re-use:

```
quarkus.log.handler.splunk.include-exception=true
quarkus.log.handler.splunk.include-logger-name=true
quarkus.log.handler.splunk.include-thread-name=true
```

We'd need a generic mechanism to override the fields name (also useful for MDC key remapping)
Another structured handler uses something like `quarkus.log.console.json.key-overrides=level=severity` (weird syntax)

Reply:
Yes feels like clients of distributer logging solution provider have to be highly configurable to match the different company policies